qiskit_ibm_runtime/options/options.py

```diff
@@ -155,6 +169,56 @@ def _get_program_inputs(options: dict) -> dict:
                 inputs[key] = options[key]
         return inputs
 
+    @staticmethod
+    def _validate_program_inputs(
+        options: dict, primitive: PrimitiveType = "Sampler", backend: IBMBackend = None
```

Comment:
Instead of passing in the primitive name and have branches, perhaps it makes more sense for `Sampler` and `Estimator` classes to have their own `_validate_options` methods that call a general method (i.e. this one) and then do their own primitive-specific validation.

Reply:
Good idea!